src/main/java/org/folio/services/adjusment/AdjustmentsService.java

```diff
@@ -190,23 +231,55 @@ private List<InvoiceLine> applyAmountTypeProratedAdjustments(Adjustment adjustme
       }
 
       Adjustment proratedAdjustment = prepareAdjustmentForLine(adjustment);
-      proratedAdjustment.setValue(amount.getNumber()
-        .doubleValue());
-      if (addAdjustmentToLine(line, proratedAdjustment)) {
+      proratedAdjustment.setValue(amount.getNumber().doubleValue());
+      if (addAdjustmentToLine(line.getAdjustments(), proratedAdjustment)) {
         updatedLines.add(line);
       }
     }
 
     return updatedLines;
   }
 
+  private List<InvoiceLine> applyProratedAmountTypeIncludedInAdjustments(Adjustment adjustment, List<InvoiceLine> lines,
+                                                                         CurrencyUnit currencyUnit) {
+    List<InvoiceLine> updatedLines = new ArrayList<>();
+    for (InvoiceLine line : lines) {
+      if (invoiceLineWasAdjustedById(adjustment, line)) {
+        continue;
+      }
+      Adjustment preparedAdjustment = prepareIncludedInAdjustmentValue(adjustment, currencyUnit, line);
+      if (addAdjustmentToLine(line.getAdjustments(), preparedAdjustment)) {
+        updatedLines.add(line);
+      }
+    }
+    return updatedLines;
+  }
+
+  private boolean invoiceLineWasAdjustedById(Adjustment adjustment, InvoiceLine line) {
+    return Objects.nonNull(adjustment.getId()) && line.getAdjustments().stream()
+      .map(Adjustment::getAdjustmentId)
+      .filter(Objects::nonNull)
+      .anyMatch(lineAdjustmentId -> lineAdjustmentId.equals(adjustment.getId()));
```

Comment:
small suggestion can be separate to variables to faster read
line.getAdjustments().stream()
      .map(Adjustment::getAdjustmentId)
      .filter(Objects::nonNull)
      .anyMatch(lineAdjustmentId -> lineAdjustmentId.equals(adjustment.getId()));